### PR TITLE
Add POM settings for deploying/releasing

### PR DESCRIPTION
**Pull request by v1v**
### What

Enable the settings that are required to run the release automation in the CI.

This automation relies in the existing release automation defined for the apm-agent-java project

### Why

We want to automate the release and binary distribution for this project

#### DryRun

```bash
$ ./mvnw \
   -DdryRun=true
   release:prepare \
   --settings .ci/settings.xml \
   -Darguments="-DskipTests --settings .ci/settings.xml" \
   --batch-mode
```

<details><summary>Expand to view the console output</summary>
<p>

produced:

```
[INFO] Scanning for projects...
[WARNING] 
[WARNING] Some problems were encountered while building the effective model for co.elastic.opentelemetry.contrib:opentelemetry-maven-extension:jar:0.1.0-beta-3-SNAPSHOT
[WARNING] 'build.plugins.plugin.version' for org.apache.maven.plugins:maven-javadoc-plugin is missing. @ line 240, column 21
[WARNING] 
[WARNING] It is highly recommended to fix these problems because they threaten the stability of your build.
[WARNING] 
[WARNING] For this reason, future Maven versions might no longer support building such malformed projects.
[WARNING] 
[INFO] 
[INFO] ---< co.elastic.opentelemetry.contrib:opentelemetry-maven-extension >---
[INFO] Building OpenTelemetry Maven Extension 0.1.0-beta-3-SNAPSHOT
[INFO] --------------------------------[ jar ]---------------------------------
[INFO] 
[INFO] --- maven-release-plugin:2.5.3:prepare (default-cli) @ opentelemetry-maven-extension ---
[INFO] Resuming release from phase 'scm-check-modifications'
[INFO] Verifying that there are no local modifications...
[INFO]   ignoring changes on: **/pom.xml.releaseBackup, **/pom.xml.next, **/pom.xml.tag, **/pom.xml.branch, **/release.properties, **/pom.xml.backup
[INFO] Executing: /bin/sh -c cd /Users/vmartinez/work/src/github.com/elastic/opentelemetry-maven-extension && git rev-parse --show-toplevel
[INFO] Working directory: /Users/vmartinez/work/src/github.com/elastic/opentelemetry-maven-extension
[INFO] Executing: /bin/sh -c cd /Users/vmartinez/work/src/github.com/elastic/opentelemetry-maven-extension && git status --porcelain .
[INFO] Working directory: /Users/vmartinez/work/src/github.com/elastic/opentelemetry-maven-extension
[INFO] Checking dependencies and plugins for snapshots ...
[INFO] Transforming 'OpenTelemetry Maven Extension'...
[INFO] Not generating release POMs
[INFO] Executing preparation goals - since this is simulation mode it is running against the original project, not the rewritten ones
[INFO] Executing goals 'clean verify'...
[INFO] [INFO] Scanning for projects...
[INFO] [WARNING] 
[INFO] [WARNING] Some problems were encountered while building the effective model for co.elastic.opentelemetry.contrib:opentelemetry-maven-extension:jar:0.1.0-beta-3-SNAPSHOT
[INFO] [WARNING] 'build.plugins.plugin.version' for org.apache.maven.plugins:maven-javadoc-plugin is missing. @ line 240, column 21
[INFO] [WARNING] 
[INFO] [WARNING] It is highly recommended to fix these problems because they threaten the stability of your build.
[INFO] [WARNING] 
[INFO] [WARNING] For this reason, future Maven versions might no longer support building such malformed projects.
[INFO] [WARNING] 
[INFO] [INFO] 
[INFO] [INFO] ---< co.elastic.opentelemetry.contrib:opentelemetry-maven-extension >---
[INFO] [INFO] Building OpenTelemetry Maven Extension 0.1.0-beta-3-SNAPSHOT
[INFO] [INFO] --------------------------------[ jar ]---------------------------------
[INFO] [INFO] 
[INFO] [INFO] --- maven-clean-plugin:2.5:clean (default-clean) @ opentelemetry-maven-extension ---
[INFO] [INFO] 
[INFO] [INFO] --- maven-resources-plugin:2.6:resources (default-resources) @ opentelemetry-maven-extension ---
[INFO] [INFO] Using 'UTF-8' encoding to copy filtered resources.
[INFO] [INFO] skip non existing resourceDirectory /Users/vmartinez/work/src/github.com/elastic/opentelemetry-maven-extension/src/main/resources
[INFO] [INFO] 
[INFO] [INFO] --- maven-compiler-plugin:3.8.0:compile (default-compile) @ opentelemetry-maven-extension ---
[INFO] [INFO] Changes detected - recompiling the module!
[INFO] [INFO] Compiling 11 source files to /Users/vmartinez/work/src/github.com/elastic/opentelemetry-maven-extension/target/classes
[INFO] [INFO] 
[INFO] [INFO] --- plexus-component-metadata:2.1.0:generate-metadata (default) @ opentelemetry-maven-extension ---
[INFO] [INFO] Discovered 5 component descriptors(s)
[INFO] [INFO] 
[INFO] [INFO] --- maven-resources-plugin:2.6:testResources (default-testResources) @ opentelemetry-maven-extension ---
[INFO] [INFO] Using 'UTF-8' encoding to copy filtered resources.
[INFO] [INFO] skip non existing resourceDirectory /Users/vmartinez/work/src/github.com/elastic/opentelemetry-maven-extension/src/test/resources
[INFO] [INFO] 
[INFO] [INFO] --- maven-compiler-plugin:3.8.0:testCompile (default-testCompile) @ opentelemetry-maven-extension ---
[INFO] [INFO] Changes detected - recompiling the module!
[INFO] [INFO] Compiling 3 source files to /Users/vmartinez/work/src/github.com/elastic/opentelemetry-maven-extension/target/test-classes
[INFO] [INFO] 
[INFO] [INFO] --- plexus-component-metadata:2.1.0:generate-test-metadata (default) @ opentelemetry-maven-extension ---
[INFO] [INFO] 
[INFO] [INFO] --- maven-surefire-plugin:2.12.4:test (default-test) @ opentelemetry-maven-extension ---
[INFO] [INFO] Tests are skipped.
[INFO] [INFO] 
[INFO] [INFO] --- maven-jar-plugin:2.4:jar (default-jar) @ opentelemetry-maven-extension ---
[INFO] [INFO] Building jar: /Users/vmartinez/work/src/github.com/elastic/opentelemetry-maven-extension/target/opentelemetry-maven-extension-0.1.0-beta-3-SNAPSHOT.jar
[INFO] [INFO] 
[INFO] [INFO] --- maven-shade-plugin:3.2.4:shade (default) @ opentelemetry-maven-extension ---
[INFO] [INFO] Including org.codehaus.plexus:plexus-component-annotations:jar:2.1.0 in the shaded jar.
[INFO] [INFO] Including com.google.code.findbugs:jsr305:jar:3.0.1 in the shaded jar.
[INFO] [INFO] Including io.opentelemetry:opentelemetry-api:jar:1.5.0 in the shaded jar.
[INFO] [INFO] Including io.opentelemetry:opentelemetry-context:jar:1.5.0 in the shaded jar.
[INFO] [INFO] Including io.opentelemetry:opentelemetry-sdk:jar:1.5.0 in the shaded jar.
[INFO] [INFO] Including io.opentelemetry:opentelemetry-sdk-common:jar:1.5.0 in the shaded jar.
[INFO] [INFO] Including io.opentelemetry:opentelemetry-semconv:jar:1.5.0-alpha in the shaded jar.
[INFO] [INFO] Including io.opentelemetry:opentelemetry-sdk-trace:jar:1.5.0 in the shaded jar.
[INFO] [INFO] Including io.opentelemetry:opentelemetry-api-metrics:jar:1.5.0-alpha in the shaded jar.
[INFO] [INFO] Including io.opentelemetry:opentelemetry-exporter-otlp:jar:1.5.0 in the shaded jar.
[INFO] [INFO] Including io.grpc:grpc-netty-shaded:jar:1.39.0 in the shaded jar.
[INFO] [INFO] Including com.google.guava:guava:jar:30.1.1-jre in the shaded jar.
[INFO] [INFO] Including com.google.guava:failureaccess:jar:1.0.1 in the shaded jar.
[INFO] [INFO] Including com.google.guava:listenablefuture:jar:9999.0-empty-to-avoid-conflict-with-guava in the shaded jar.
[INFO] [INFO] Including org.checkerframework:checker-qual:jar:3.8.0 in the shaded jar.
[INFO] [INFO] Including com.google.j2objc:j2objc-annotations:jar:1.3 in the shaded jar.
[INFO] [INFO] Including com.google.errorprone:error_prone_annotations:jar:2.4.0 in the shaded jar.
[INFO] [INFO] Including io.perfmark:perfmark-api:jar:0.23.0 in the shaded jar.
[INFO] [INFO] Including io.grpc:grpc-core:jar:1.39.0 in the shaded jar.
[INFO] [INFO] Including com.google.code.gson:gson:jar:2.8.6 in the shaded jar.
[INFO] [INFO] Including com.google.android:annotations:jar:4.1.1.4 in the shaded jar.
[INFO] [INFO] Including org.codehaus.mojo:animal-sniffer-annotations:jar:1.19 in the shaded jar.
[INFO] [INFO] Including io.opentelemetry:opentelemetry-exporter-otlp-trace:jar:1.5.0 in the shaded jar.
[INFO] [INFO] Including io.grpc:grpc-api:jar:1.39.0 in the shaded jar.
[INFO] [INFO] Including io.grpc:grpc-context:jar:1.39.0 in the shaded jar.
[INFO] [INFO] Including io.grpc:grpc-protobuf:jar:1.39.0 in the shaded jar.
[INFO] [INFO] Including com.google.api.grpc:proto-google-common-protos:jar:2.0.1 in the shaded jar.
[INFO] [INFO] Including io.grpc:grpc-protobuf-lite:jar:1.39.0 in the shaded jar.
[INFO] [INFO] Including io.grpc:grpc-stub:jar:1.39.0 in the shaded jar.
[INFO] [INFO] Including com.google.protobuf:protobuf-java:jar:3.17.2 in the shaded jar.
[INFO] [INFO] Including io.opentelemetry:opentelemetry-exporter-logging:jar:1.5.0 in the shaded jar.
[INFO] [INFO] Including io.opentelemetry:opentelemetry-sdk-metrics:jar:1.5.0-alpha in the shaded jar.
[INFO] [INFO] Including io.opentelemetry:opentelemetry-exporter-otlp-common:jar:1.5.0 in the shaded jar.
[INFO] [INFO] Including io.opentelemetry:opentelemetry-proto:jar:1.5.0-alpha in the shaded jar.
[INFO] [INFO] Replacing original artifact with shaded artifact.
[INFO] [INFO] Replacing /Users/vmartinez/work/src/github.com/elastic/opentelemetry-maven-extension/target/opentelemetry-maven-extension-0.1.0-beta-3-SNAPSHOT.jar with /Users/vmartinez/work/src/github.com/elastic/opentelemetry-maven-extension/target/opentelemetry-maven-extension-0.1.0-beta-3-SNAPSHOT-shaded.jar
[INFO] [INFO] Dependency-reduced POM written at: /Users/vmartinez/work/src/github.com/elastic/opentelemetry-maven-extension/dependency-reduced-pom.xml
[INFO] [INFO] 
[INFO] [INFO] >>> spotbugs-maven-plugin:4.2.0:check (default) > :spotbugs @ opentelemetry-maven-extension >>>
[INFO] [INFO] 
[INFO] [INFO] --- spotbugs-maven-plugin:4.2.0:spotbugs (spotbugs) @ opentelemetry-maven-extension ---
[INFO] [INFO] Fork Value is true
[INFO] [INFO] Done SpotBugs Analysis....
[INFO] [INFO] 
[INFO] [INFO] <<< spotbugs-maven-plugin:4.2.0:check (default) < :spotbugs @ opentelemetry-maven-extension <<<
[INFO] [INFO] 
[INFO] [INFO] 
[INFO] [INFO] --- spotbugs-maven-plugin:4.2.0:check (default) @ opentelemetry-maven-extension ---
[INFO] [INFO] BugInstance size is 0
[INFO] [INFO] Error size is 0
[INFO] [INFO] No errors/warnings found
[INFO] [INFO] ------------------------------------------------------------------------
[INFO] [INFO] BUILD SUCCESS
[INFO] [INFO] ------------------------------------------------------------------------
[INFO] [INFO] Total time:  10.061 s
[INFO] [INFO] Finished at: 2021-09-03T14:12:24+01:00
[INFO] [INFO] ------------------------------------------------------------------------
[INFO] Full run would be commit 1 files with message: '[maven-release-plugin] prepare release v0.1.0-beta-3'
[INFO] Full run would be tagging working copy /Users/vmartinez/work/src/github.com/elastic/opentelemetry-maven-extension with label: 'v0.1.0-beta-3'
[INFO] Transforming 'OpenTelemetry Maven Extension'...
[INFO] Not removing release POMs
[INFO] Executing completion goals - since this is simulation mode it is running against the original project, not the rewritten ones
[INFO] Full run would be commit 1 files with message: '[maven-release-plugin] prepare for next development iteration'
[INFO] Release preparation simulation complete.
[INFO] ------------------------------------------------------------------------
[INFO] BUILD SUCCESS
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  11.924 s
[INFO] Finished at: 2021-09-03T14:12:24+01:00
[INFO] ------------------------------------------------------------------------
```

</p>
</details>

#### release:perform

*NOTE*: Failed since it could not push to the upstream repo

```bash
$ ./mvnw \
   release:prepare \
   release:perform \
   --settings .ci/settings.xml \
   -Darguments="-DskipTests --settings .ci/settings.xml" \
   --batch-mode
```

<details><summary>Expand to view the console output</summary>
<p>

```
./mvnw release:prepare release:perform --settings .ci/settings.xml -Darguments="-DskipTests --settings .ci/settings.xml" --batch-mode    
[INFO] Scanning for projects...
[WARNING] 
[WARNING] Some problems were encountered while building the effective model for co.elastic.opentelemetry.contrib:opentelemetry-maven-extension:jar:0.1.0-beta-3-SNAPSHOT
[WARNING] 'build.plugins.plugin.version' for org.apache.maven.plugins:maven-javadoc-plugin is missing. @ line 240, column 21
[WARNING] 
[WARNING] It is highly recommended to fix these problems because they threaten the stability of your build.
[WARNING] 
[WARNING] For this reason, future Maven versions might no longer support building such malformed projects.
[WARNING] 
[INFO] 
[INFO] ---< co.elastic.opentelemetry.contrib:opentelemetry-maven-extension >---
[INFO] Building OpenTelemetry Maven Extension 0.1.0-beta-3-SNAPSHOT
[INFO] --------------------------------[ jar ]---------------------------------
[INFO] 
[INFO] --- maven-release-plugin:2.5.3:prepare (default-cli) @ opentelemetry-maven-extension ---
[INFO] Verifying that there are no local modifications...
[INFO]   ignoring changes on: **/pom.xml.releaseBackup, **/pom.xml.next, **/pom.xml.tag, **/pom.xml.branch, **/release.properties, **/pom.xml.backup
[INFO] Executing: /bin/sh -c cd /Users/vmartinez/work/src/github.com/elastic/opentelemetry-maven-extension && git rev-parse --show-toplevel
[INFO] Working directory: /Users/vmartinez/work/src/github.com/elastic/opentelemetry-maven-extension
[INFO] Executing: /bin/sh -c cd /Users/vmartinez/work/src/github.com/elastic/opentelemetry-maven-extension && git status --porcelain .
[INFO] Working directory: /Users/vmartinez/work/src/github.com/elastic/opentelemetry-maven-extension
[INFO] Checking dependencies and plugins for snapshots ...
[INFO] Transforming 'OpenTelemetry Maven Extension'...
[INFO] Not generating release POMs
[INFO] Executing goals 'clean verify'...
[INFO] [INFO] Scanning for projects...
[INFO] [WARNING] 
[INFO] [WARNING] Some problems were encountered while building the effective model for co.elastic.opentelemetry.contrib:opentelemetry-maven-extension:jar:0.1.0-beta-3
[INFO] [WARNING] 'build.plugins.plugin.version' for org.apache.maven.plugins:maven-javadoc-plugin is missing. @ line 240, column 21
[INFO] [WARNING] 
[INFO] [WARNING] It is highly recommended to fix these problems because they threaten the stability of your build.
[INFO] [WARNING] 
[INFO] [WARNING] For this reason, future Maven versions might no longer support building such malformed projects.
[INFO] [WARNING] 
[INFO] [INFO] 
[INFO] [INFO] ---< co.elastic.opentelemetry.contrib:opentelemetry-maven-extension >---
[INFO] [INFO] Building OpenTelemetry Maven Extension 0.1.0-beta-3
[INFO] [INFO] --------------------------------[ jar ]---------------------------------
[INFO] [INFO] 
[INFO] [INFO] --- maven-clean-plugin:2.5:clean (default-clean) @ opentelemetry-maven-extension ---
[INFO] [INFO] Deleting /Users/vmartinez/work/src/github.com/elastic/opentelemetry-maven-extension/target
[INFO] [INFO] 
[INFO] [INFO] --- maven-resources-plugin:2.6:resources (default-resources) @ opentelemetry-maven-extension ---
[INFO] [INFO] Using 'UTF-8' encoding to copy filtered resources.
[INFO] [INFO] skip non existing resourceDirectory /Users/vmartinez/work/src/github.com/elastic/opentelemetry-maven-extension/src/main/resources
[INFO] [INFO] 
[INFO] [INFO] --- maven-compiler-plugin:3.8.0:compile (default-compile) @ opentelemetry-maven-extension ---
[INFO] [INFO] Changes detected - recompiling the module!
[INFO] [INFO] Compiling 11 source files to /Users/vmartinez/work/src/github.com/elastic/opentelemetry-maven-extension/target/classes
[INFO] [INFO] 
[INFO] [INFO] --- plexus-component-metadata:2.1.0:generate-metadata (default) @ opentelemetry-maven-extension ---
[INFO] [INFO] Discovered 5 component descriptors(s)
[INFO] [INFO] 
[INFO] [INFO] --- maven-resources-plugin:2.6:testResources (default-testResources) @ opentelemetry-maven-extension ---
[INFO] [INFO] Using 'UTF-8' encoding to copy filtered resources.
[INFO] [INFO] skip non existing resourceDirectory /Users/vmartinez/work/src/github.com/elastic/opentelemetry-maven-extension/src/test/resources
[INFO] [INFO] 
[INFO] [INFO] --- maven-compiler-plugin:3.8.0:testCompile (default-testCompile) @ opentelemetry-maven-extension ---
[INFO] [INFO] Changes detected - recompiling the module!
[INFO] [INFO] Compiling 3 source files to /Users/vmartinez/work/src/github.com/elastic/opentelemetry-maven-extension/target/test-classes
[INFO] [INFO] 
[INFO] [INFO] --- plexus-component-metadata:2.1.0:generate-test-metadata (default) @ opentelemetry-maven-extension ---
[INFO] [INFO] 
[INFO] [INFO] --- maven-surefire-plugin:2.12.4:test (default-test) @ opentelemetry-maven-extension ---
[INFO] [INFO] Tests are skipped.
[INFO] [INFO] 
[INFO] [INFO] --- maven-jar-plugin:2.4:jar (default-jar) @ opentelemetry-maven-extension ---
[INFO] [INFO] Building jar: /Users/vmartinez/work/src/github.com/elastic/opentelemetry-maven-extension/target/opentelemetry-maven-extension-0.1.0-beta-3.jar
[INFO] [INFO] 
[INFO] [INFO] --- maven-shade-plugin:3.2.4:shade (default) @ opentelemetry-maven-extension ---
[INFO] [INFO] Including org.codehaus.plexus:plexus-component-annotations:jar:2.1.0 in the shaded jar.
[INFO] [INFO] Including com.google.code.findbugs:jsr305:jar:3.0.1 in the shaded jar.
[INFO] [INFO] Including io.opentelemetry:opentelemetry-api:jar:1.5.0 in the shaded jar.
[INFO] [INFO] Including io.opentelemetry:opentelemetry-context:jar:1.5.0 in the shaded jar.
[INFO] [INFO] Including io.opentelemetry:opentelemetry-sdk:jar:1.5.0 in the shaded jar.
[INFO] [INFO] Including io.opentelemetry:opentelemetry-sdk-common:jar:1.5.0 in the shaded jar.
[INFO] [INFO] Including io.opentelemetry:opentelemetry-semconv:jar:1.5.0-alpha in the shaded jar.
[INFO] [INFO] Including io.opentelemetry:opentelemetry-sdk-trace:jar:1.5.0 in the shaded jar.
[INFO] [INFO] Including io.opentelemetry:opentelemetry-api-metrics:jar:1.5.0-alpha in the shaded jar.
[INFO] [INFO] Including io.opentelemetry:opentelemetry-exporter-otlp:jar:1.5.0 in the shaded jar.
[INFO] [INFO] Including io.grpc:grpc-netty-shaded:jar:1.39.0 in the shaded jar.
[INFO] [INFO] Including com.google.guava:guava:jar:30.1.1-jre in the shaded jar.
[INFO] [INFO] Including com.google.guava:failureaccess:jar:1.0.1 in the shaded jar.
[INFO] [INFO] Including com.google.guava:listenablefuture:jar:9999.0-empty-to-avoid-conflict-with-guava in the shaded jar.
[INFO] [INFO] Including org.checkerframework:checker-qual:jar:3.8.0 in the shaded jar.
[INFO] [INFO] Including com.google.j2objc:j2objc-annotations:jar:1.3 in the shaded jar.
[INFO] [INFO] Including com.google.errorprone:error_prone_annotations:jar:2.4.0 in the shaded jar.
[INFO] [INFO] Including io.perfmark:perfmark-api:jar:0.23.0 in the shaded jar.
[INFO] [INFO] Including io.grpc:grpc-core:jar:1.39.0 in the shaded jar.
[INFO] [INFO] Including com.google.code.gson:gson:jar:2.8.6 in the shaded jar.
[INFO] [INFO] Including com.google.android:annotations:jar:4.1.1.4 in the shaded jar.
[INFO] [INFO] Including org.codehaus.mojo:animal-sniffer-annotations:jar:1.19 in the shaded jar.
[INFO] [INFO] Including io.opentelemetry:opentelemetry-exporter-otlp-trace:jar:1.5.0 in the shaded jar.
[INFO] [INFO] Including io.grpc:grpc-api:jar:1.39.0 in the shaded jar.
[INFO] [INFO] Including io.grpc:grpc-context:jar:1.39.0 in the shaded jar.
[INFO] [INFO] Including io.grpc:grpc-protobuf:jar:1.39.0 in the shaded jar.
[INFO] [INFO] Including com.google.api.grpc:proto-google-common-protos:jar:2.0.1 in the shaded jar.
[INFO] [INFO] Including io.grpc:grpc-protobuf-lite:jar:1.39.0 in the shaded jar.
[INFO] [INFO] Including io.grpc:grpc-stub:jar:1.39.0 in the shaded jar.
[INFO] [INFO] Including com.google.protobuf:protobuf-java:jar:3.17.2 in the shaded jar.
[INFO] [INFO] Including io.opentelemetry:opentelemetry-exporter-logging:jar:1.5.0 in the shaded jar.
[INFO] [INFO] Including io.opentelemetry:opentelemetry-sdk-metrics:jar:1.5.0-alpha in the shaded jar.
[INFO] [INFO] Including io.opentelemetry:opentelemetry-exporter-otlp-common:jar:1.5.0 in the shaded jar.
[INFO] [INFO] Including io.opentelemetry:opentelemetry-proto:jar:1.5.0-alpha in the shaded jar.
[INFO] [INFO] Replacing original artifact with shaded artifact.
[INFO] [INFO] Replacing /Users/vmartinez/work/src/github.com/elastic/opentelemetry-maven-extension/target/opentelemetry-maven-extension-0.1.0-beta-3.jar with /Users/vmartinez/work/src/github.com/elastic/opentelemetry-maven-extension/target/opentelemetry-maven-extension-0.1.0-beta-3-shaded.jar
[INFO] [INFO] Dependency-reduced POM written at: /Users/vmartinez/work/src/github.com/elastic/opentelemetry-maven-extension/dependency-reduced-pom.xml
[INFO] [INFO] 
[INFO] [INFO] >>> spotbugs-maven-plugin:4.2.0:check (default) > :spotbugs @ opentelemetry-maven-extension >>>
[INFO] [INFO] 
[INFO] [INFO] --- spotbugs-maven-plugin:4.2.0:spotbugs (spotbugs) @ opentelemetry-maven-extension ---
[INFO] [INFO] Fork Value is true
[INFO] [INFO] Done SpotBugs Analysis....
[INFO] [INFO] 
[INFO] [INFO] <<< spotbugs-maven-plugin:4.2.0:check (default) < :spotbugs @ opentelemetry-maven-extension <<<
[INFO] [INFO] 
[INFO] [INFO] 
[INFO] [INFO] --- spotbugs-maven-plugin:4.2.0:check (default) @ opentelemetry-maven-extension ---
[INFO] [INFO] BugInstance size is 0
[INFO] [INFO] Error size is 0
[INFO] [INFO] No errors/warnings found
[INFO] [INFO] ------------------------------------------------------------------------
[INFO] [INFO] BUILD SUCCESS
[INFO] [INFO] ------------------------------------------------------------------------
[INFO] [INFO] Total time:  9.095 s
[INFO] [INFO] Finished at: 2021-09-03T14:19:34+01:00
[INFO] [INFO] ------------------------------------------------------------------------
[INFO] Checking in modified POMs...
[INFO] Executing: /bin/sh -c cd /Users/vmartinez/work/src/github.com/elastic/opentelemetry-maven-extension && git add -- pom.xml
[INFO] Working directory: /Users/vmartinez/work/src/github.com/elastic/opentelemetry-maven-extension
[INFO] Executing: /bin/sh -c cd /Users/vmartinez/work/src/github.com/elastic/opentelemetry-maven-extension && git rev-parse --show-toplevel
[INFO] Working directory: /Users/vmartinez/work/src/github.com/elastic/opentelemetry-maven-extension
[INFO] Executing: /bin/sh -c cd /Users/vmartinez/work/src/github.com/elastic/opentelemetry-maven-extension && git status --porcelain .
[INFO] Working directory: /Users/vmartinez/work/src/github.com/elastic/opentelemetry-maven-extension
[INFO] Executing: /bin/sh -c cd /Users/vmartinez/work/src/github.com/elastic/opentelemetry-maven-extension && git commit --verbose -F /var/folders/hn/j6gds1qx1f5bysfn0qp0t03c0000gn/T/maven-scm-1144940823.commit pom.xml
[INFO] Working directory: /Users/vmartinez/work/src/github.com/elastic/opentelemetry-maven-extension
[INFO] Executing: /bin/sh -c cd /Users/vmartinez/work/src/github.com/elastic/opentelemetry-maven-extension && git symbolic-ref HEAD
[INFO] Working directory: /Users/vmartinez/work/src/github.com/elastic/opentelemetry-maven-extension
[INFO] Executing: /bin/sh -c cd /Users/vmartinez/work/src/github.com/elastic/opentelemetry-maven-extension && git push git@github.com:elastic/opentelemetry-maven-extension.git refs/heads/feature/add-settings.xml:refs/heads/feature/add-settings.xml
[INFO] Working directory: /Users/vmartinez/work/src/github.com/elastic/opentelemetry-maven-extension
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  13.086 s
[INFO] Finished at: 2021-09-03T14:19:37+01:00
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-release-plugin:2.5.3:prepare (default-cli) on project opentelemetry-maven-extension: Unable to commit files
[ERROR] Provider message:
[ERROR] The git-push command failed.
[ERROR] Command output:
[ERROR] ERROR: Permission to elastic/opentelemetry-maven-extension.git denied to v1v.
[ERROR] fatal: Could not read from remote repository.
[ERROR] 
[ERROR] Please make sure you have the correct access rights
[ERROR] and the repository exists.
[ERROR] -> [Help 1]
[ERROR] 
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR] 
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoFailureException
```

```diff
diff --git a/pom.xml b/pom.xml
index 9c38c17..2c6ab83 100644
--- a/pom.xml
+++ b/pom.xml
@@ -8,7 +8,7 @@
     -->
     <groupId>co.elastic.opentelemetry.contrib</groupId>
     <artifactId>opentelemetry-maven-extension</artifactId>
-    <version>0.1.0-beta-3-SNAPSHOT</version>
+    <version>0.1.0-beta-3</version>
     <packaging>jar</packaging>
     <name>OpenTelemetry Maven Extension</name>
     <description>
@@ -284,7 +284,7 @@
         <connection>scm:git:git@github.com:elastic/opentelemetry-maven-extension.git</connection>
         <developerConnection>scm:git:git@github.com:elastic/opentelemetry-maven-extension.git</developerConnection>
         <url>https://github.com/elastic/opentelemetry-maven-extension</url>
-        <tag>HEAD</tag>
+        <tag>v0.1.0-beta-3</tag>
     </scm>
     <issueManagement>
         <system>GitHub Issues</system>
```

</p>
</details>

### CODE REVIEW

**Review by cyrille-leclerc — APPROVED** [review]

File: .ci/settings.xml [hunk 1]
```diff
@@ -0,0 +1,29 @@
+<settings xmlns="http://maven.apache.org/SETTINGS/1.1.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
```
Comment (v1v) [inline .ci/settings.xml:1]: See https://github.com/elastic/apm-agent-java/blob/master/.ci/settings.xml

File: pom.xml [hunk 2]
```diff
@@ -150,6 +150,31 @@
             </dependency>
         </dependencies>
     </dependencyManagement>
+
+    <profiles>
```
Comment (v1v) [inline pom.xml:154]: See https://github.com/elastic/apm-agent-java/blob/7752081468a96cb4702a3c1e29cf481e0e159754/pom.xml#L232-L253

File: pom.xml [hunk 3]
```diff
@@ -219,7 +244,34 @@
                     <source>8</source>
                 </configuration>
             </plugin>
+            <plugin>
```
Comment (v1v) [inline pom.xml:247]: See https://github.com/elastic/apm-agent-java/blob/7752081468a96cb4702a3c1e29cf481e0e159754/pom.xml#L258-L276

File: pom.xml [hunk 4]
```diff
@@ -219,7 +244,34 @@
...
+                    <releaseProfiles>gpg</releaseProfiles>
+                    <tagNameFormat>v@{project.version}</tagNameFormat>
+                </configuration>
+            </plugin>
+            <plugin>
+                <artifactId>maven-deploy-plugin</artifactId>
+            </plugin>
         </plugins>
+        <pluginManagement>
+            <plugins>
```
Comment (v1v) [inline pom.xml:260]: See https://github.com/elastic/apm-agent-java/blob/7752081468a96cb4702a3c1e29cf481e0e159754/pom.xml#L439-L450

File: pom.xml [hunk 5]
```diff
@@ -229,9 +281,11 @@
         </repository>
     </distributionManagement>
     <scm>
-        <connection>scm:git:https://github.com/elastic/opentelemetry-maven-extension.git</connection>
-      <tag>HEAD</tag>
-  </scm>
+        <connection>scm:git:git@github.com:elastic/opentelemetry-maven-extension.git</connection>
```
Comment (v1v) [inline pom.xml:284]: See https://github.com/elastic/apm-agent-java/blob/7752081468a96cb4702a3c1e29cf481e0e159754/pom.xml#L29-L34

**Pull request merged**